src/terraformpy/resource_collections.py

```diff
@@ -53,6 +54,12 @@ def __init__(self, *args, **kwargs):
             super(ResourceCollection, self).__init__(*args, **kwargs)
             return
 
+        # there are still some places in underlying schematics stuff that
+        # invoke model constructors in the traditional way, but without
+        # context. get_mock_object() is one of these cases
+        if len(kwargs) == 0 and len(args) == 1 and isinstance(args[0], dict):
+            kwargs = args[0]
```

Comment:
I know we don't use `args` anywhere else right now, to ensure we leave `args` in a "correct" state I would prefer we slice index 0 off of it:

```python
kwargs = args[0]
args = args[1:]
```

Reply:
Or even: `args = tuple()`